client/src/mid.rs

```diff
@@ -75,8 +114,50 @@ pub struct State {
     views: SlotMap<ViewKey, View>,
     /// connected url
     url: String,
+    client: ClientWithMiddleware,
     /// Connection status
     status: bool,
+    mid_event_sender: Sender<MidEvent>,
+}
+
+/// reverts some state given a server error
+#[derive(Debug)]
+enum RevertError {
+    Task(TaskKey),
+    Tasks(Vec<TaskKey>),
+    Filter(ViewKey),
+}
+
+#[derive(Debug)]
+/// Events to be handled by the middleware
+pub enum MidEvent {
+    #[allow(private_interfaces)]
+    ServerResponse(Result<Box<dyn ServerResponse>, RevertError>),
+    StateEvent(StateEvent), // events to be handled by the ui
+}
+impl State {
+    // handles MidEvents, so long as MidEvent is not of variant StateEvent
+    pub fn handle_mid_event(&mut self, event: MidEvent) -> color_eyre::Result<()> {
```

Comment:
This is where the events are handled after the HTTP requests get executed right?